src/Umbraco.Core/Persistence/Repositories/IUserRepository.cs

```diff
@@ -81,6 +81,39 @@ public interface IUserRepository : IReadWriteQueryRepository<int, IUser>
     /// </returns>
     IUser? GetByUsername(string username, bool includeSecurityData);
 
+    /// <summary>
+    /// Gets a user by username for upgrade purposes, this will only return a result if the current runtime state is upgrade.
+    /// </summary>
+    /// <remarks>
+    /// This only resolves the minimum amount of fields required to authorize for an upgrade.
+    /// We need this to be able to add new columns to the user table.
+    /// </remarks>
+    /// <param name="username">The username to find the user by.</param>
+    /// <returns>An uncached <see cref="IUser"/> instance.</returns>
+    IUser? GetForUpgradeByUsername(string username) => GetByUsername(username, false);
+
+    /// <summary>
+    /// Gets a user by email for upgrade purposes, this will only return a result if the current runtime state is upgrade.
+    /// </summary>
+    /// <remarks>
+    /// This only resolves the minimum amount of fields required to authorize for an upgrade.
+    /// We need this to be able to add new columns to the user table.
+    /// </remarks>
+    /// <param name="email">The email to find the user by.</param>
+    /// <returns>An uncached <see cref="IUser"/> instance.</returns>
+    IUser? GetForUpgradeByEmail(string email) => throw new NotImplementedException();
```

Comment:
Still think this implementation is better than throwing. Hopefully nobody will ever hit that code :)

```suggestion
    IUser? GetForUpgradeByEmail(string email) => GetMany().FirstOrDefault(x=>x.Email == email);
```

Reply:
Very good point 👍